lib/routes/content.js

```diff
@@ -284,18 +284,38 @@ const routes = application => {
      * @param {object} next - Express callback
     */
     app.post(`/${config.content.base}/:type/${config.content.actions.save}`, multipartMiddleware, (req, res, next) => {
-      const referrer = _.get(req.session, 'referrer') || `/${config.content.base}/${req.params.type.toLowerCase()}`;
+      const referrer = _.get(req.session, 'referrer') || req.get('Referrer');
       let audits;
       let publishable = false;
       let check = 'publish';
       let files = [];
-
-      _.unset(req.session, 'referrer');
+      let source = {};
 
       if (req.body.submit === config.content.actions.new) {
         check = 'save';
       }
 
+      // determine data source from referrer
+      if (_.includes(referrer, '/add')) {
+        source.type = 'add';
+      }
+      else if (_.includes(referrer, '/edit')) {
+        source = {
+          type: 'edit',
+          id: _.get(req.session, `form.content.edit[${req.params.type.toLowerCase()}].id`, ''),
```

Comment:
no. this has different needs. ID can only come from the session because the `save` url does not have any variables in it. Value is doublechecked below (line 333)